docs/sources.md

```diff
@@ -0,0 +1,246 @@
+Other sources are considered as inherently directory-based sources, like git repositories.
+Depending on the source type, the behavior of certain things may be different, depending on the target implementation (e.g. mariner2, jammy, windows, etc).
+Sources are injected into the root path of the build environment using the name of the source.
+
+Ideally the content of a source is platform agnostic (e.g. no platform specific binaries).
+These sources are used to create source packages for the target platform, such as an srpm or a debian dsc.
+
+## Top-level source configuration
+
+For all source types, you can specify the following top-level configuration:
+
+- `path`: The path to extract from the source type
+- `includes`: A list of glob patterns to include from the source
+- `excludes`: A list of glob patterns to exclude from the source
+
+## Source Types
+
+### Git
+
+Git sources fetch a git repository at a specific commit.
+You can use either an SSH style git URL or an HTTPS style git URL.
+
+For SSH style git URLs, if the client (such as the docker CLI) has provided
+access to an SSH agent, that agent will be used to authenticate with the git
+server.
+
+```yaml
+sources:
+  someSource1:
+    git:
+      # This uses an SSH style git URL.
+      url: git@github.com:myOrg/myRepo.git
+      commit: 1234567890abcdef
+  someSource2:
+    git:
+      # This uses an HTTPS style git URL.
+      url: https://github.com/myOrg/myRepo.git
+      commit: 1234567890abcdef
+```
+
+By default, Dalec will discard the `.git` directory when fetching a git source.
+You can override this behavior by setting `keepGitDir: true` in the git configuration.
+
+Git repositories are considered to be "directory" sources.
+
+
+### HTTP
+
+HTTP sources fetch a file from an HTTP URL.
+HTTP content is not verified by digest today, but it is in the roadmap.
+
+```yaml
+sources:
+  someSource1:
+    http:
+      # No Digest verification
+      url: https://example.com/someFile.txt
+```
+
+The HTTP source type is considered to be a "file" source.
+
+### Build context
+
+Clients provide a build context to Dalec.
+As an  example, here is how the Docker client provides a build context to Dalec:
+
+```bash
+$ docker build <some args> .
+```
+
+In this case the `.`, or current directory, is the build context.
+Dalec is able to use the build context as a source:
+
+```yaml
+sources:
+  someSource:
+    context: {}
+```
+
+Note the empty brackets.
+This is an unfortunate syntax requirement to not have `context` considered as a nil value.
+This is the equivelent of the following:
+
+```yaml
+sources:
+  someSource:
+    context:
+      name: "context"
+```
+
+Where `name: "context"`, not to be confused with the source type `context`, is named by convention by the docker CLI.
+Additionally contexts can be passed in from the docker cli: `docker build --build-context <name>=<path>`.
+The `<name>` would be the name to use in your yaml to access it.
```

Comment:
for this referring to the name of the context. Could we maybe add an example of how it could be used in the yaml?

Reply:
The example is using a name, I included one that does not include the name.